stripe/src/main/java/com/stripe/android/net/PollingNetworkHandler.java

```diff
@@ -0,0 +1,187 @@
+
+        mTimeoutMs = timeOutMs == null ? 10000L : timeOutMs.longValue();
+
+        mUiHandler = new Handler(Looper.getMainLooper()) {
+            int delayMs = INITIAL_DELAY_MS;
+            boolean terminated = false;
+
+            @Override
+            public void handleMessage(Message msg) {
+                super.handleMessage(msg);
+                if (terminated) {
+                    return;
+                }
+
+                switch (msg.what) {
+                    case SUCCESS:
+                        callback.onSuccess();
+                        break;
+                    case PENDING:
+                        delayMs *= POLLING_MULTIPLIER;
+                        callback.onRetry(delayMs);
+                        mNetworkHandler.sendEmptyMessage(delayMs);
+                        break;
+                    case FAILURE:
+                        terminated = true;
+                        callback.onError(new PollingFailedException(AUTHORIZATION_FAILED, false));
+                        break;
+                    case EXPIRED:
+                        terminated = true;
+                        callback.onError(new PollingFailedException(POLLING_EXPIRED, true));
+                        break;
+                    case ERROR:
+                        terminated = true;
+                        callback.onError((StripeException) msg.obj);
+                        break;
+                    default:
+                        break;
+                }
+            }
+        };
+
+        HandlerThread handlerThread = null;
+        if (!mIsInSingleThreadMode) {
+            handlerThread = new HandlerThread("Network Handler Thread");
+            handlerThread.start();
+        }
+
+        final Looper looper = mIsInSingleThreadMode
+                ? Looper.getMainLooper()
+                : handlerThread.getLooper();
+        mNetworkHandler = new Handler(looper) {
+            boolean terminated = false;
+
+            @Override
+            public void handleMessage(Message msg) {
+                super.handleMessage(msg);
+                if (terminated) {
+                    return;
+                }
+
+                if (msg.what >= 0) {
+                    postDelayed(pollRunnable, msg.what);
+                } else {
+                    terminated = true;
+                    if (!mIsInSingleThreadMode) {
+                        looper.quit();
+                    }
+                    mUiHandler.removeMessages(SUCCESS);
+                    mUiHandler.removeMessages(PENDING);
+                    mUiHandler.removeMessages(FAILURE);
+                    removeCallbacks(pollRunnable);
+                }
+            }
+        };
+    }
+
+    public void start() {
+        mNetworkHandler.post(pollRunnable);
+        mNetworkHandler.sendEmptyMessageDelayed(EXPIRED, mTimeoutMs);
+        mUiHandler.sendEmptyMessageDelayed(EXPIRED, mTimeoutMs);
+    }
+
+    interface SourceRetriever {
```

Comment:
How do I test a static method that calls out to our network? Code to an otherwise useless interface!